src/entitlements-client-factory.ts

```diff
@@ -1,22 +1,26 @@
-import { ClientConfiguration } from './client-configuration';
+import { ClientConfiguration, FallbackConfiguration } from './client-configuration';
 import axios from 'axios';
 import { EntitlementsClient } from './entitlements.client';
 import { LoggingClient, SimpleLoggingClient } from './logging';
 import { OpaQueryClient } from './opa-queries';
 import { ConfigurationInputIsMissingException } from './exceptions/configuration-input-is-missing.exception';
+import { ConfigurationInputIsInvalidException } from './exceptions/configuration-input-is-invalid.exception';
 
 export class EntitlementsClientFactory {
 	public static create(configuration: ClientConfiguration): EntitlementsClient {
 		if (!configuration.pdpHost) {
 			throw new ConfigurationInputIsMissingException('pdpHost is required');
 		}
+		if (configuration.timeout !== undefined && configuration.timeout <= 0) {
```

Comment:
why not use default? i think we should have default timeout (unless debug mode)

Reply:
There's a default timeout from axios.
If we want to change that, pass a timeout